### DISCUSSION TITLE
HasOne relationship shows trashed resources

### DISCUSSION THREAD

**Discussion by Hannoma** [laravel/nova-issues#4120] [Q&A] [answered]
- Laravel Version: 8.83.9
- Nova Version: 4.2.5
- PHP Version: 8.1.3

### Description:
In my user resource, there is a HasOne relationship to accesses. The accesses are using the soft-deleted trait. In Nova, the attached soft-deleted access is shown in the user detail page. In our case, this is not expected behavior and this behavior deviates from the HasOne relationship behavior of laravel/eloquent. Because in that case, soft-deleted models are not concerned. In addition, there is no method available like for the BelongsTo field to filter out trashed models.
![image](https://user-images.githubusercontent.com/7955676/165283115-76686e97-a8d4-40e1-8ab2-cf19fa6e79f8.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:
`Models/User.php`
```php
public function access(): HasOne
{
    return $this->hasOne(Access::class);
}
```

`Nova/User.php`
```php
public function fields(NovaRequest $request): array
{
    // More fields here
    HasOne::make('Access')->exceptOnForms(),
}
```

`Models/Access.php`
```php
public function user(): BelongsTo
{
    return $this->belongsTo(User::class)->withTrashed();
}
```

`Nova/Access.php`
```php
public function fields(NovaRequest $request): array
{
    // More fields here
    BelongsTo::make('User')
                ->exceptOnForms(),
}
```

**Comment by crynobone** [comment, marked as answer]
> In our case, this is not expected behavior and this behavior deviates from the HasOne relationship behavior of laravel/eloquent.

Detail query is executed without global scope in order for Nova to still access soft deleted `Access` model.

Now, it seem that the relationship should have been **User has one of many Access**, where you can define it via:

```php
// Models/User.php
public function profile()
{
    return $this->hasOne(Access::class)->ofMany()->latest()->withoutTrashed();
}
```

```php
// Nova/User.php
public function fields(NovaRequest $request)
{
    return [
        HasOne::ofMany('Profile'),
    ];
}
```

**Reply by Hannoma** [reply]
It seems like that even this works:
```php
// Models/User.php
public function access(): HasOne
{
    return $this->hasOne(Access::class)->ofMany();
}
```

**Reply by wajdijurry** [reply]
@Hannoma Short working answer. Thank you
